### PR TITLE
Amelia/trajectory vehicle

### PR DESCRIPTION
**Pull request by AmeliaY07**
Amelia added trajectory vehicle agent

### CODE REVIEW

File: gym_minigrid/agents/TrajectoryVehicle.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        id,
+        trajectory: list,
+        topLeft: Tuple[int, int],
+        bottomRight: Tuple[int, int],
+        direction: int,
+        maxSpeed: float,
+        speed: float,
+        inRoad: int,
+        inLane: int,
+        objectType="Vehicle"
```
Comment (taoruihuang) [inline gym_minigrid/agents/TrajectoryVehicle.py]: does not need as constructor parameter

File: gym_minigrid/agents/TrajectoryVehicle.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            super().__init__(
+                inRoad=int,
+                inLane=int,
+                id=id,
+                topLeft=topLeft,
+                bottomRight=bottomRight,
+                direction=direction,
+                maxSpeed=maxSpeed,
+                speed=speed,
+                objectType=objectType
```
Comment (taoruihuang) [inline gym_minigrid/agents/TrajectoryVehicle.py]: can remove this line

File: gym_minigrid/envs/pedestrian/TwoLaneRoadEnv.py [hunk 3]
```diff
@@ -23,6 +25,7 @@ def __init__(
         self,
         pedAgents: List[PedAgent]=[],
         vehicleAgents: List[Vehicle]=[],
+        trajectoryVehicleAgents: List[TrajectoryVehicle]=[],
```
Comment (taoruihuang) [inline gym_minigrid/envs/pedestrian/TwoLaneRoadEnv.py]: there can be a separate list of trajectory agents, environment does not know what type of vehicle, remove this list

File: gym_minigrid/envs/pedestrian/TwoLaneRoadEnv.py [hunk 4]
```diff
@@ -60,6 +67,16 @@ def addVehicleAgent(self, agent: Vehicle):
         # subscribe to events here
         super().subscribe(EnvEvent.stepParallel2, agent.go)
 
+    def addTrajectoryVehicleAgents(self, agents: List[TrajectoryVehicle]):
```
Comment (taoruihuang) [inline gym_minigrid/envs/pedestrian/TwoLaneRoadEnv.py]: remove these two methods, change agent.go to agent.parallel1

**Review by taoruihuang — DISMISSED** [review]

File: gym_minigrid/agents/TrajectoryVehicle.py (hunk 1 above)
Comment (AmeliaY07) [reply gym_minigrid/agents/TrajectoryVehicle.py]: deleted

**Review by adhocmaster — APPROVED** [review]

**Pull request merged**